splitio/storage/pluggable.py

```diff
@@ -657,3 +678,167 @@ def clear(self):
+
+    def _reset_config_tags(self):
+        """Reset config tags."""
+        with self._lock:
+            self._config_tags = []
+
+    def add_config_tag(self, tag):
+        """
+        Record tag string.
+
+        :param tag: tag to be added
+        :type tag: str
+        """
+        with self._lock:
+            if len(self._config_tags) < MAX_TAGS:
+                self._config_tags.append(tag)
+
+    def record_config(self, config, extra_config):
+        """
+        initilize telemetry objects
+
+        :param config: factory configuration parameters
+        :type config: Dict
+        :param extra_config: any extra configs
+        :type extra_config: Dict
+        """
+        self._tel_config.record_config(config, extra_config)
+
+    def pop_config_tags(self):
+        """Get and reset configs."""
+        with self._lock:
+            tags = self._config_tags
+            self._reset_config_tags()
+            return tags
+
+    def push_config_stats(self):
+        """push config stats to storage."""
+        self._pluggable_adapter.set(self._telemetry_config_key + "::" + self._sdk_metadata, str(self._format_config_stats()))
+
+    def _format_config_stats(self):
+        """format only selected config stats to json"""
+        config_stats = self._tel_config.get_stats()
+        return json.dumps({
+            'aF': config_stats['aF'],
+            'rF': config_stats['rF'],
+            'sT': config_stats['sT'],
+            'oM': config_stats['oM'],
+            't': self.pop_config_tags()
+        })
+
+    def record_active_and_redundant_factories(self, active_factory_count, redundant_factory_count):
+        """
+        Record active and redundant factories.
+
+        :param active_factory_count: active factory count
+        :type active_factory_count: int
+        :param redundant_factory_count: redundant factory count
+        :type redundant_factory_count: int
+        """
+        self._tel_config.record_active_and_redundant_factories(active_factory_count, redundant_factory_count)
+
+    def record_latency(self, method, latency):
+        """
+        record latency data
+
+        :param method: method name
+        :type method: string
+        :param latency: latency
+        :type latency: int64
+        """
+        self._method_latencies.add_latency(method, latency)
+        latencies = self._method_latencies.pop_all()['methodLatencies']
+        values = latencies[method.value]
+        total_keys = 0
+        bucket_number = 0
+        for bucket in values:
+            if bucket > 0:
+                latency_key = self._telemetry_latencies_key + '::' + self._sdk_metadata + '/' + method.value + '/' + str(bucket_number)
+                result = self._pluggable_adapter.increment(latency_key, bucket)
+                self.expire_keys(latency_key, self._TELEMETRY_KEY_DEFAULT_TTL, 1, result)
+                total_keys += 1
+            bucket_number = bucket_number + 0
```

Comment:
???????

Reply:
This looks like a bug in redis class, using your code it is not needed though, will apply it to redis class as well